crates/sui-framework/sources/governance/stake.move

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module sui::stake {
+    use std::option::{Self, Option};
+    use sui::balance::Balance;
+    use sui::id::VersionedID;
+    use sui::locked_coin;
+    use sui::sui::SUI;
+    use sui::transfer;
+    use sui::tx_context::{Self, TxContext};
+    use sui::epoch_time_lock::EpochTimeLock;
+    use sui::epoch_time_lock;
+    use sui::balance;
+    use sui::math;
+
+    friend sui::sui_system;
+    friend sui::validator;
+
+    /// A custodial stake object holding the staked SUI coin.
+    struct Stake has key {
+        id: VersionedID,
+        /// The staked SUI tokens.
+        balance: Balance<SUI>,
+        /// The epoch until which the staked coin is locked. If the stake
+        /// comes from a Coin<SUI>, this field is None. If it comes from a LockedCoin<SUI>, this
+        /// field will record the original lock expiration epoch, to be used when unstaking.
+        coin_locked_until_epoch: Option<EpochTimeLock>,
```

Comment:
Yeah I see why the naming can be confusing. I don’t want to name it as something like `stake_locked_until_epoch` though, because that sounds like the bonding period for unstaking. This field is for storing the unlock epoch of the original `LockedCoin` that was passed in while creating the stake. This is needed so we can put the balance back into a `LockedCoin` with the same unlock epoch when unstaking. Do you think `original_coin_locked_until_epoch` is better? What would you say is the best name here? @lxfind

Reply:
just call it `locked_until_epoch` or `unlock_epoch`